salt/apache/_common.conf

```diff
@@ -0,0 +1,62 @@
+# vi: ft=apache
+
+{% if https == 'yes' or https == 'force' %}
+{% set myportlist = [ '80', '443' ] %}
+{% else %}
+{% set myportlist = [ '80' ] %}
+{% endif %}
+
+# servername = {{ servername }}
+# serveraliases = {{ serveraliases }}
+# https = {{ https }}
+# myportlist = {{ myportlist }}
+
+{% for myport in myportlist %}
+
+{% if myport == '443' %}
+<IfModule mod_ssl.c>
+{% set myhttprotocol='https' %}
+{% else %}
+{% set myhttprotocol='http' %}
+{% endif %}
+<VirtualHost *:{{ myport }}>
+
+    ServerName {{ servername }}
+    {% for myalias in serveraliases %}
+    ServerAlias {{ myalias }}
+    {% endfor %}
+    ServerAdmin code@opendataservices.coop
+    {% if myport=='80' and https=='force' %}
+        Redirect / https://{{ servername }}
+    {% else %}
+
+        {% if myport=='443' %}
+        SSLEngine on
+
+        # "Intermediate" configuration
+        SSLProtocol             all -SSLv2 -SSLv3
+        SSLCipherSuite          ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA
+        SSLHonorCipherOrder     on
+        SSLCompression          off
+        SSLOptions +StrictRequire
+
+        SSLCertificateFile      /etc/letsencrypt/live/{{ servername }}/cert.pem
```

Comment:
This is a problem atm, because the apache config is invalid if this doesn't exist - and it doesn't exist until letsencrypt has spoken to our apache server - catch 22.